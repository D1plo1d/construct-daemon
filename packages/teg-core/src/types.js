/* config */
export { default as Config, MockConfig } from './config/types/Config'
export * as AxisTypeEnum from './config/types/AxisTypeEnum'
export * as ComponentTypeEnum from './config/types/components/ComponentTypeEnum'

/* jobQueue */
export { default as JobHistoryEvent } from './jobQueue/types/JobHistoryEvent'
export { default as Job, MockJob } from './jobQueue/types/Job'
export { default as JobFile, MockJobFile } from './jobQueue/types/JobFile'
export * as TaskStatusEnum from './jobQueue/types/TaskStatusEnum'
export { default as Task, MockTask } from './jobQueue/types/Task'

/* printer */

/* devices */
export { default as Device } from './devices/types/Device'
export * as DeviceTypeEnum from './devices/types/DeviceTypeEnum'
