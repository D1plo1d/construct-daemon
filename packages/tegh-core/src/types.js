/* config */
export Config, { MockConfig } from './config/types/Config'
export * as AxisTypeEnum from './config/types/AxisTypeEnum'

/* jobQueue */
export * as JobHistoryTypeEnum from './jobQueue/types/JobHistoryTypeEnum'

export JobHistoryEvent from './jobQueue/types/JobHistoryEvent'
export JobQueueGraphQL from './jobQueue/types/JobQueue.graphql.js'
export Job, { MockJob } from './jobQueue/types/Job'
export JobGraphQL from './jobQueue/types/Job.graphql.js'
export JobFile, { MockJobFile } from './jobQueue/types/JobFile'

/* printer */
export PrinterGraphQL from './printer/types/Printer.graphql.js'

/* spool */
export * as PriorityEnum from './spool/types/PriorityEnum'
export * as TaskStatusEnum from './spool/types/TaskStatusEnum'
export * as ComponentTypeEnum from './config/types/ComponentTypeEnum'

export Task, { MockTask } from './spool/types/Task'
export TaskGraphQL from './spool/types/Task.graphql.js'
