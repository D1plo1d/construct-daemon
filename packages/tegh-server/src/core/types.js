/* jobQueue */
export { default as Job } from './jobQueue/types/Job'
export { default as JobFile } from './jobQueue/types/JobFile'

import * as JobStatusEnum from './jobQueue/types/JobStatusEnum'
export { JobStatusEnum }

/* spool */
export { default as Task } from './spool/types/Task'

import * as PriorityEnum from './spool/types/PriorityEnum'
export { PriorityEnum }

import * as TaskStatusEnum from './spool/types/TaskStatusEnum'
export { TaskStatusEnum }
