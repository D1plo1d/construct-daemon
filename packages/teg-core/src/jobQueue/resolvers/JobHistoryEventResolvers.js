const JobHistoryEventResolvers = {
  JobHistoryEvent: {
    type: source => source.type.replace('/jobQueue/JobHistory/', ''),
  },
}

export default JobHistoryEventResolvers
