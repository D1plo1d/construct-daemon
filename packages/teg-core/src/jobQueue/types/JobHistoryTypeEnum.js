import { List } from 'immutable'

/*
 * START_EVENT: the first time a print is attempted or a re-print attempt after
 * a CANCEL_EVENT or ERROR_EVENT.
 */
export const SPOOL_PRINT = '/jobQueue/JobHistory/SPOOL_PRINT'
export const START_PRINT = '/jobQueue/JobHistory/START_PRINT'
export const CANCEL_PRINT = '/jobQueue/JobHistory/CANCEL_PRINT'
export const PRINT_ERROR = '/jobQueue/JobHistory/PRINT_ERROR'
export const FINISH_PRINT = '/jobQueue/JobHistory/FINISH_PRINT'

export const SPOOLED_TYPES = List([
  SPOOL_PRINT,
  START_PRINT,
])

const JobHistoryTypeEnum = List([
  SPOOL_PRINT,
  START_PRINT,
  CANCEL_PRINT,
  PRINT_ERROR,
  FINISH_PRINT,
])


export default JobHistoryTypeEnum
