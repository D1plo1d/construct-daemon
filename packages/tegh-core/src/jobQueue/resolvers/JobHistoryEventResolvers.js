const JobHistoryEventResolvers = {
  type: source => source.type.replace('/jobQueue/JobHistory/', ''),
}

export default JobHistoryEventResolvers
