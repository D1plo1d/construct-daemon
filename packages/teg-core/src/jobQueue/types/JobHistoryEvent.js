import uuid from 'uuid'
import { Record } from 'immutable'

import JobHistoryTypeEnum from './JobHistoryTypeEnum'

const JobHistoryEventRecord = Record({
  id: null,
  jobID: null,
  jobFileID: null,
  taskID: null,
  createdAt: null,
  type: null,
})

const JobHistoryEvent = (attrs) => {
  const { type } = attrs

  if (JobHistoryTypeEnum.includes(type) === false) {
    throw new Error(`Invalid Job History type: ${attrs.type}`)
  }

  return JobHistoryEventRecord({
    id: uuid.v4(),
    createdAt: new Date().toISOString(),
    ...attrs,
  })
}

export default JobHistoryEvent
