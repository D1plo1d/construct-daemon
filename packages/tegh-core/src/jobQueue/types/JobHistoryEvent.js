import uuid from 'uuid/v4'
import { Record } from 'immutable'

import JobHistoryTypeEnum, { STARTED } from './JobHistoryTypeEnum'

const JobHistoryEventRecord = Record({
  id: null,
  jobFileID: null,
  printerID: null,
  createdAt: null,
  type: null,
})

const JobHistoryEvent = (attrs) => {
  const { type, printerID } = attrs

  if (JobHistoryTypeEnum.includes(type) === false) {
    throw new Error(`Invalid Job History type: ${attrs.type}`)
  }
  if (printerID == null && type === STARTED) {
    throw new Error('printerID must not be null for type: STARTED')
  }

  return JobHistoryEventRecord({
    id: uuid(),
    createdAt: new Date().toISOString(),
    ...attrs,
  })
}

export default JobHistoryEvent
