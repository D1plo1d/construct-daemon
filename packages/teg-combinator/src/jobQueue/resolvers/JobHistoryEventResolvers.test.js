import typeDefs from '@tegapp/schema'

import deterministicTestSetup from '../../util/testing/deterministicTestSetup'
import snapshotTestResolvers from '../../util/testing/snapshotTestResolvers'

import JobHistoryEvent from '../types/JobHistoryEvent'
import { FINISH_PRINT } from '../types/JobHistoryTypeEnum'

import JobHistoryEventResolvers from './JobHistoryEventResolvers'

describe('JobHistoryEventResolvers', () => {
  it('presents a consistent API', () => {
    deterministicTestSetup()

    const event = JobHistoryEvent({
      type: FINISH_PRINT,
    })


    snapshotTestResolvers({
      typeDefs,
      typeName: 'JobHistoryEvent',
      resolvers: JobHistoryEventResolvers,
      rootValue: event,
      contextValue: {},
    })
  })
})
