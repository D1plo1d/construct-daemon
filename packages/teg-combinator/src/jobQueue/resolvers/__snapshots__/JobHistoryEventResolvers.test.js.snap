// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobHistoryEventResolvers presents a consistent API 1`] = `
Object {
  "_testJobHistoryEvent": Object {
    "createdAt": "2018-01-01T00:00:00.000Z",
    "id": "MOCK_ID_0",
    "type": "FINISH_PRINT",
  },
}
`;
